lib/Conversion/TorchToTcp/Elementwise.cpp

```diff
@@ -105,6 +105,90 @@ class ConvertAtenMulOp : public OpConversionPattern<AtenMulTensorOp> {
   }
 };
 
+class ConvertAtenBatchNormOp : public OpConversionPattern<AtenBatchNormOp> {
+public:
+  using OpConversionPattern<AtenBatchNormOp>::OpConversionPattern;
+  using OpAdaptor = typename AtenBatchNormOp::Adaptor;
+
+  LogicalResult
+  matchAndRewrite(AtenBatchNormOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Value input = adaptor.getInput();
+    RankedTensorType inputType = input.getType().dyn_cast<RankedTensorType>();
+
+    Value weight = adaptor.getWeight();
+    RankedTensorType weightType = weight.getType().dyn_cast<RankedTensorType>();
+
+    Value bias = adaptor.getBias();
+    RankedTensorType biasType = bias.getType().dyn_cast<RankedTensorType>();
+
+    Value runningMean = adaptor.getRunningMean();
+    RankedTensorType runningMeanType =
+        runningMean.getType().dyn_cast<RankedTensorType>();
+
+    Value runningVar = adaptor.getRunningVar();
+    RankedTensorType runningVarType =
+        runningVar.getType().dyn_cast<RankedTensorType>();
+
+    RankedTensorType resultType =
+        OpConversionPattern<AtenBatchNormOp>::getTypeConverter()
+            ->convertType(op.getType())
+            .template cast<RankedTensorType>();
+
+    if (!inputType || !weightType || !biasType || !runningMeanType ||
+        !runningVarType || !resultType)
+      return rewriter.notifyMatchFailure(
+          op, "Only Ranked Tensor types are supported in TCP");
+
+    assert(runningMeanType.getNumElements() != 0 &&
+           runningVarType.getNumElements() != 0);
```

Comment:
This *can* happen in valid torch dialect MLIR, right? If so, an assert should not be used. I would use a `notifyMatchFailure`

Reply:
Thanks, yes I fixed this from an earlier comment but it probably didn't update while you were reviewing. This is fixed now.